mail_tracking/models/mail_tracking_email.py

```diff
@@ -92,13 +92,21 @@ class MailTrackingEmail(models.Model):
         string="Tracking events", comodel_name='mail.tracking.event',
         inverse_name='tracking_email_id', readonly=True)
 
+    @api.multi
+    def write(self, vals):
+        if vals.get('state') in self.env['mail.message']._failed_states:
+            self.mapped('mail_message_id').write({
+                'mail_tracking_needs_action': True,
+            })
+        super().write(vals)
+
     @api.model
     def email_is_bounced(self, email):
         if not email:
             return False
         res = self._email_last_tracking_state(email)
-        return res and res[0].get('state', '') in ['rejected', 'error',
-                                                   'spam', 'bounced']
+        return res and res[0].get('state', '') in \
+            self.env['mail.message']._failed_states
```

Comment:
This isn't correct for soft-bounce errors, wich are temporary (like full mailbox, recipient mail server down, etc).
We're even questioning the 'error' event as it triggers in some cases wich are not the recipients fault like smtp server errors, etc.